### PR TITLE
Docs: Add changelog entries for the 1.0.0, 1.1.0, and 1.2.0 releases

### PR DESCRIPTION
**Pull request by brycelelbach**

### CODE REVIEW

File: docs/releases/changelog.md [hunk 1]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+### Other Enhancements
+
+- C++17 support.
+- NVCC + GCC 9 support.
+- NVCC + Clang 9 support.
+- Build with warnings-as-errors.
+
+### Issues Fixed
+
+- TODO
```
Comment (brycelelbach) [inline docs/releases/changelog.md:141]: @griwes please review the changelog (`git log 1.0.0..1.1.0`) and fill this information in. You can suggest changes to the PR through GitHub's web interface.

File: docs/releases/changelog.md [hunk 2]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+It also introduces `<cuda/std/type_traits>`, `<cuda/std/cassert>`,
+`<cuda/std/cfloat>`, `<cuda/std/cstddef>`, and `<cuda/std/cstdint>`.
+
+### New Features
 
-TODO: Warn about Debian package bug.
+- `<cuda/[std/]atomic>`:
+  - `cuda::thread_scope`: TODO
+  - `cuda::atomic<T, Scope>`: TODO
+  - `cuda::std::atomic<T>`: TODO
```
Comment (brycelelbach) [inline docs/releases/changelog.md:156]: @ogiroux please come up with an appropriate summary for these items. Look at some of the other entries in this file to get an idea of the style and length I'm look for. You can suggest changes to the PR through GitHub's web interface.

File: docs/releases/changelog.md [hunk 3]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+It brings C++ atomics to CUDA: `<cuda/[std/]atomic>`.
+It also introduces `<cuda/std/type_traits>`, `<cuda/std/cassert>`,
+`<cuda/std/cfloat>`, `<cuda/std/cstddef>`, and `<cuda/std/cstdint>`.
+
+### New Features
 
-TODO: Warn about Debian package bug.
+- `<cuda/[std/]atomic>`:
+  - `cuda::thread_scope`: TODO
+  - `cuda::atomic<T, Scope>`: TODO
```
Comment (brycelelbach) [inline docs/releases/changelog.md:155]: @ogiroux please come up with an appropriate summary for these items. Look at some of the other entries in this file to get an idea of the style and length I'm look for. You can suggest changes to the PR through GitHub's web interface.

File: docs/releases/changelog.md [hunk 4]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+  entire system.
+It brings C++ atomics to CUDA: `<cuda/[std/]atomic>`.
+It also introduces `<cuda/std/type_traits>`, `<cuda/std/cassert>`,
+`<cuda/std/cfloat>`, `<cuda/std/cstddef>`, and `<cuda/std/cstdint>`.
+
+### New Features
 
-TODO: Warn about Debian package bug.
+- `<cuda/[std/]atomic>`:
+  - `cuda::thread_scope`: TODO
```
Comment (brycelelbach) [inline docs/releases/changelog.md:154]: @ogiroux please come up with an appropriate summary for these items. Look at some of the other entries in this file to get an idea of the style and length I'm look for. You can suggest changes to the PR through GitHub's web interface.

File: docs/releases/changelog.md [hunk 5]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+  `<cuda/std/functional>`.
+
+### ABI Breaking Changes
+
+- ABI version 2 has been introduced and is now the default.
+  A new ABI version was introduced because it is our policy to do so in every
+    major CUDA toolkit release.
+  ABI version 1 is no longer supported.
+
+### Breaking Changes
```
Comment (brycelelbach) [inline docs/releases/changelog.md:108]: @griwes @ogiroux did we break anything else in 1.1.0 / 11.0?

File: docs/releases/changelog.md [hunk 6]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+- `<cuda/std/functional>`: common function objects, such as `cuda::std::plus`,
+    `cuda::std::minus`, etc.
+  `cuda::std::function`, `cuda::std::bind`, `cuda::std::hash`, and
+    `cuda::std::reference_wrapper` are omitted.
+
+### Other Enhancements
+
+- C++17 support.
+- NVCC + GCC 9 support.
+- NVCC + Clang 9 support.
```
Comment (brycelelbach) [inline docs/releases/changelog.md:136]: @griwes did we add support for any other platforms in 1.1.0 / 11.0?

File: docs/releases/changelog.md [hunk 7]
```diff
@@ -27,19 +46,125 @@ supported with NVCC + MSVC. It also adds documentation.
...
+  This facility is NOT for transferring data between threads or transferring
+    data between host and device; it is not a `cudaMemcpyAsync` replacement or
+    abstraction.
+  It uses `cuda::[std::]barrier`s objects to synchronize the copies.
+- `<cuda/std/functional>`: common function objects, such as `cuda::std::plus`,
+    `cuda::std::minus`, etc.
+  `cuda::std::function`, `cuda::std::bind`, `cuda::std::hash`, and
+    `cuda::std::reference_wrapper` are omitted.
+
+### Other Enhancements
```
Comment (brycelelbach) [inline docs/releases/changelog.md:132]: @griwes did we make any other enhancements in 1.1.0 / 11.0?

File: docs/releases/changelog.md (hunk 5 above)
Comment (griwes) [reply docs/releases/changelog.md:108]: I don't think the one listed below is really a break - it never really worked in the first place.

File: docs/releases/changelog.md (hunk 7 above)
Comment (griwes) [reply docs/releases/changelog.md:132]: We upgraded to a newer version of the upstream. We also fixed a bug where we had a non-inline function (`__cuda_memcmp`) caused redefinition errors.

File: docs/releases/changelog.md (hunk 6 above)
Comment (griwes) [reply docs/releases/changelog.md:136]: Yes - NVRTC with no more dependencies on the host headers.

**Review by griwes — APPROVED** [review]

File: docs/releases/changelog.md (hunk 5 above)
Comment (brycelelbach) [reply docs/releases/changelog.md:108]: I'm going to leave it in; we didn't previously say it was unsupported.

File: docs/releases/changelog.md (hunk 6 above)
Comment (brycelelbach) [reply docs/releases/changelog.md:136]: Added.

**Pull request merged**